dockerfiles/hpcc/templates/thor.yaml

```diff
@@ -0,0 +1,128 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ printf "%s-slave" .name | quote }}
+spec:
+  replicas: {{ $thor.numSlaves }}
+  selector:
+    matchLabels:
+      run: {{ printf "%s-slave" .name | quote }}
+  template:
+    metadata:
+      labels:
+        run: {{ printf "%s-slave" .name | quote }}
+    spec:
+      {{ if $.Values.global.singleNode | default false -}}
+      initContainers:
+      # This is a bit of a hack, to ensure that the persistent storage mounted for dllserver
+      # is writable. This is not something we would want to do if using anything other than
+      # local storage (which is only sensible on single-node systems).
+      - name: volume-mount-hack
+        image: busybox
+        command: ["sh", "-c", "chown -R 999:1000 /var/lib/HPCCSystems/hpcc-data"]
+        volumeMounts:
+        - mountPath: "/var/lib/HPCCSystems/hpcc-data"
+          name: mythorstorage
+      {{- end }}
+      containers:
+        - name: {{ printf "%s-slave" .name | quote }}
+          args: [
+                  {{ include "hpcc.utils.configArg" . }},
+                  {{ include "hpcc.utils.daliArg" $ }},
+                  --slaveport={{ $thor.slaveport }},
+                  --master={{ printf "%s-svc:%v" .name $thor.masterport }}
+                ]
+{{ include "hpcc.utils.addImageAttrs" (dict "root" $ "imagename" "thorslave") | indent 10 }}
+          ports:
+            - containerPort: {{ $thor.slaveport }}
+{{ include "hpcc.utils.addVolumeMounts" . | indent 10 }}
+          - mountPath: /var/lib/HPCCSystems/hpcc-data
+            name: mythorstorage
+        {{ if $thor.startDafilesrv | default false -}}
+        - name: {{ printf "%s-slave-dafilesrv" .name | quote }}
+{{ include "hpcc.utils.addImageAttrs" (dict "root" $ "imagename" "thorslave") | indent 10 }}
+          command: ["/opt/HPCCSystems/bin/dafilesrv"]
+          ports:
+            - containerPort: 7100
+          volumeMounts:
+            - mountPath: /var/lib/HPCCSystems/hpcc-data
+              name: mythorstorage
+        {{- end }}
+{{ include "hpcc.utils.addVolumes" . | indent 6 }}
+      - name: mythorstorage
+        persistentVolumeClaim:
+          claimName: {{ .name }}-pvc
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ printf "%s-svc" .name | quote }}
+spec:
+  ports:
+  - port: {{ $thor.masterport }}
+    protocol: TCP
+    targetPort: {{ $thor.masterport }}
+  selector:
+    run: {{ .name | quote }}
+  type: ClusterIP
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ printf "%s-pv" .name | quote }}
+  labels:
+    type: local
+spec:
+  storageClassName: manual
+  capacity:
+    storage: {{ $thor.storageSize }}
+  accessModes:
+    - ReadWriteMany
+  persistentVolumeReclaimPolicy: Retain
+  hostPath:
+    path: {{ printf "%s/hpcc-data" $.Values.global.hostMountPath }}
+    type: DirectoryOrCreate
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ printf "%s-pvc" .name | quote }}
```

Comment:
I might be inclined to use printf once at top to save the required string into a variable - just for clarity and consistency with how it was done in roxie.yaml

Reply:
yep, will do for this and a few others.